### PR TITLE
[mbstring] add return value to mb_parse_str (#351)

### PR DESCRIPTION
**Pull request by webber2k6**
This pull request fixes the TypeError from #351.

### CODE REVIEW

File: src/Mbstring/bootstrap80.php [hunk 1]
```diff
@@ -51,7 +51,7 @@ function mb_detect_encoding(?string $string, array|string|null $encodings = null
     function mb_detect_order(array|string|null $encoding = null): array|bool { return p\Mbstring::mb_detect_order((string) $encoding); }
 }
 if (!function_exists('mb_parse_str')) {
-    function mb_parse_str(?string $string, &$result = []): bool { parse_str((string) $string, $result); }
+    function mb_parse_str(?string $string, &$result = []): bool { parse_str((string) $string, $result); return !empty($result); }
```
Comment (nicolas-grekas) [inline src/Mbstring/bootstrap80.php:54]: `return (bool) $result;` would be enough

**Review by nicolas-grekas — COMMENTED** [review]
Thanks for the PR.
Can you please add a test case in `tests/Mbstring/MbstringTest.php`?

**Comment by nicolas-grekas** [comment]
`src/Mbstring/bootstrap.php` has the same issue btw (the test case would spot it ;) )

**Comment by webber2k6** [comment]
> `src/Mbstring/bootstrap.php` has the same issue btw (the test case would spot it ;) )

The function in `src/Mbstring/bootstrap.php` has no return type annotation.

**Comment by nicolas-grekas** [comment]
> The function in src/Mbstring/bootstrap.php has no return type annotation.

sure, but the native function returns a bool nonetheless, so we have to align the behavior of the polyfil.

File: src/Mbstring/bootstrap.php [hunk 2]
```diff
@@ -55,7 +55,7 @@ function mb_detect_encoding($string, $encodings = null, $strict = false) { retur
     function mb_detect_order($encoding = null) { return p\Mbstring::mb_detect_order($encoding); }
 }
 if (!function_exists('mb_parse_str')) {
-    function mb_parse_str($string, &$result = []) { parse_str($string, $result); }
+    function mb_parse_str($string, &$result = []): bool { parse_str($string, $result); return (bool) $result; }
```
Comment (nicolas-grekas) [inline src/Mbstring/bootstrap.php:58]: the return type should be removed: the native function doesn't have one until PHP8, see https://3v4l.org/m49P0

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @webber2k6.

**Pull request merged**